Tater Box can show a Potato in place of unfound Taters

Shnupbups opened the pull request:
Can be toggled by clicking the Poisonous Potato in the bottom row, unfound taters are hidden by default,

Oh, also made a Tater Box with every Tater already inside available in the creative inventory

Shnupbups committed:
tater box shows potato for unfound taters

will add toggle for it later

haykam821 commented on src/main/java/xyz/nucleoid/extras/lobby/item/TaterBoxItem.java:
```diff
@@ -148,13 +156,13 @@ private MutableText getTitle(ItemStack stack) {
     }
 
-    private GuiElement createGuiElement(ItemStack stack, PlayerEntity user, Hand hand, ItemConvertible icon, Text name, Identifier taterId) {
+    private GuiElement createGuiElement(ItemStack stack, PlayerEntity user, Hand hand, ItemConvertible icon, Text name, Identifier taterId, boolean selectable) {
```
Since `taterId` is not applicable when `selectable` is `false`, this method should be split.

haykam821 commented on src/main/java/xyz/nucleoid/extras/lobby/item/TaterBoxItem.java:
```diff
@@ -129,14 +133,18 @@ private MutableText getTitle(ItemStack stack) {
+                        Block tater = Registry.BLOCK.get(taterId);
+                        taters.add(createGuiElement(stack, user, hand, tater, tater.getName(), taterId, true));
+                    } else {
+                        taters.add(createGuiElement(stack, user, hand, Items.POTATO, NOT_FOUND_TEXT, taterId, false));
```
Could `Items.POTATO` be a constant?

Shnupbups committed:
Merge branch '1.18' into tater-box-gui-stuff

Shnupbups committed:
add filter

Shnupbups committed:
Full Tater Box now in the creative inv

Shnupbups committed:
cleanup

+ wearing a tater now triggers the criterion for collecting it, so you dont have to find it all over again if a new advancement for it is added
